### PR TITLE
branch_type is only valid when branch_match_kind is branching_model

### PR DESCRIPTION
**Pull request by variableresistor**
Fixed issue running Add-BitbucketRepositoryBranchRestriction that caused error branch_type is only valid when branch_match_kind is branching_model

### CODE REVIEW

**Comment by variableresistor** [comment]
I can't figure out exactly how to run the Pester tests. Just running them from the test file itself doesn't work, I tried checking the contribution guidelines, and finally, I tried checking to see what the build pipeline is doing. But the script it's using is obfuscated and I can't see the script content to replicate it.

**Comment by eelkram** [comment]
@variableresistor let me see if I can shed some light onto how it works.  The tests are run from [Build.ps1](https://github.com/beyondcomputing-org/Atlassian.Bitbucket/blob/master/Build.ps1).  However, regarding testing it really only installs version 4.10.1 of Pester and then runs the tests.  The rest of the code in that script is all about generating a clean folder for publishing.  It should probably be split apart and cleaned up.

If I were to guess I would assume you probably have the latest version of Pester installed which would cause the issue.  I haven't had a need or time to refactor the tests to run on Pester 5.

Let me know if that helps or if you still have issues.

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,7 @@ _These will be removed in the next major release_
 
 ## 0.29.0
 - Updated team cmdlets to workspace while retaining old names with alias https://developer.atlassian.com/cloud/bitbucket/bitbucket-api-teams-deprecation/
+- Fixed issue running `Add-BitbucketRepositoryBranchRestriction` that caused error `branch_type is only valid when branch_match_kind is branching_model
```
Comment (eelkram) [inline CHANGELOG.md:11]: For the module to be published you will need to bump the version to 0.29.1 (bugfix).  Please update the version in the .psd1 file and then create a new entry in the CHANGELOG.md for 0.29.1.

**Review by eelkram — CHANGES_REQUESTED** [review]
Changes look good, just need to version bump to 0.29.1 and then i'll merge.

**Review by eelkram — APPROVED** [review]

**Pull request merged**